src/net.rs

```diff
@@ -299,10 +296,13 @@ impl Signable for NonceRequest {
         hasher.update(self.sign_id.to_be_bytes());
         hasher.update(self.sign_iter_id.to_be_bytes());
         hasher.update(self.message.as_slice());
-        hasher.update((self.is_taproot as u16).to_be_bytes());
-        if let Some(merkle_root) = self.merkle_root {
-            hasher.update(merkle_root);
-        }
+        hasher.update(format!("{:?}", self.signature_type));
+        /*
+            hasher.update((self.is_taproot as u16).to_be_bytes());
+            if let Some(merkle_root) = self.merkle_root {
+                hasher.update(merkle_root);
+            }
+        */
```

Comment:
I thought maybe we should do something better, like write the enum int value plus something compact for the merkle root instead of just formatting a debug string.  

What do you think?  Is the debug string okay here?

Reply:
Can't access the discriminant for non-unit enums without unsafe pointer casting, so I just went with the debug string.  I could change it to a match with branches, but the only thing that would get us is a more compact representation of the merkle root array.

Reply:
I probably would opt for matching on the enum and hashing that way. It seems a little more "correct" to do things that way, but it's more of a feeling so I'm fine with either.

Reply:
Done